src/algorithms/fardetectors/FarDetectorMLReconstruction.cc

```diff
@@ -55,6 +55,12 @@ namespace eicrecon {
     const auto [inputTracks,beamElectrons] = input;
     auto [outputFarDetectorMLTrajectories, outputFarDetectorMLTrackParameters, outputFarDetectorMLTracks] = output;
 
+    // Check if the algorithm has been initialized properly
+    if(!m_initialized){
+      debug("Initialization did not complete");
+      return;
```

Comment:
Unless you throw here, the Process will be called over and over.
```suggestion
      throw std::runtime_exception("Initialization did not complete");
```

Reply:
Will it not call process over and over anyway? I thought the behaviour we wanted was to flag the error but still return an empty collection so that the reconstruction can be carried out with the limited geometry provided. It would be nicer if upon throwing the exception in init, JANA knows not to try again.

Reply:
It will not. The exceptions in Process trickle into our JEventProcessorPODIO, where we try to print error messages only once, which is the desired effect. We could do some accounting to disable collections, but, you are right, they don't get disabled from being called currently.